### DISCUSSION TITLE
user mcap.(*indexedMessageIterator).loadChunk  Memory accumulation

### DISCUSSION THREAD

**Discussion by 97-tianyan** [foxglove/mcap#1073] [General]
When I open the MCAP file many times, if I can't consume MESSAGE in time, it will cause memory to accumulate, which will eventually lead to service OOM. The final positioning problem, I found that all the memory was occupied by Loadchunk. Is there any special feature of this method, or what needs to pay attention to?
![Uploading flamegraph.svg…]()

**Comment by defunctzombie** [comment]
Could you clarify your issue a bit more? Maybe provide some code that reproduces the issue so we can better understand what is happening. It is hard to answer the question as written because there are so many things that could be happening in your program we do not have insight into.
